include/lbann/execution_algorithms/ltfb/mutation_strategy.hpp

```diff
@@ -0,0 +1,156 @@
+{
+private:
+  std::vector<std::string> m_activation_types = {"relu", "tanh", "softmax", "elu", "leaky relu", "log softmax"};
+
+public:
+  ReplaceActivation() = default;
+
+  std::unique_ptr<lbann::Layer> make_new_activation_layer(lbann::lbann_comm& comm,
+                                                          std::string const& new_type,
+                                                          std::string const& new_name)
+  {
+    using DataType = float;
+    #ifdef LBANN_HAS_GPU
+      constexpr El::Device Dev = El::Device::GPU;
+    #else
+      constexpr El::Device Dev = El::Device::CPU;
+    #endif
+
+    std::unique_ptr<lbann::Layer> layer;
+
+    if(new_type == "relu") {
+       layer = std::make_unique<
+           lbann::relu_layer<DataType, data_layout::DATA_PARALLEL, Dev>>(&comm); 
+    } else if (new_type == "tanh") {
+       layer = std::make_unique<
+            lbann::tanh_layer<DataType, data_layout::DATA_PARALLEL, Dev>>(&comm);
+    } else if (new_type == "softmax") {
+       layer = std::make_unique<
+            lbann::softmax_layer<DataType, data_layout::DATA_PARALLEL, Dev>>(&comm, softmax_mode::INSTANCE);
+    } else if (new_type == "elu") {
+       layer = std::make_unique<
+            lbann::elu_layer<DataType, data_layout::DATA_PARALLEL, Dev>>(&comm, 1);
+    } else if (new_type == "leaky relu") {
+       layer = std::make_unique<
+            lbann::leaky_relu_layer<DataType, data_layout::DATA_PARALLEL, Dev>>(&comm, 0.01);
+    } else if (new_type == "log softmax") {
+       layer = std::make_unique<
+            lbann::log_softmax_layer<DataType, data_layout::DATA_PARALLEL, Dev>>(&comm);
+    } else {
+       LBANN_ERROR("Unknown new layer type: ", new_type);
+    }
+    layer->set_name(new_name);
+    return layer;
+  }
+
+  void mutate(model& m, const int& step)
+  {
+    std::vector<std::string> activation_layer_names;
+
+    auto& comm = *m.get_comm();
+    
+    for (int i = 0; i < m.get_num_layers(); ++i)
+    {
+       // Creating a temp string with lower case representation
+       std::string temp_type = m.get_layer(i).get_type();
+       std::transform(temp_type.begin(), temp_type.end(),
+                                         temp_type.begin(), ::tolower);    
+
+       if (std::find(m_activation_types.cbegin(), m_activation_types.cend(), temp_type)
+                                                              != m_activation_types.cend()) {
+         activation_layer_names.push_back(m.get_layer(i).get_name());
+       }
+    }
+
+    // Generate two random numbers - one for new layer type and one for old layer name
+    int type_index = fast_rand_int(get_fast_generator(), m_activation_types.size());
+    int name_index = fast_rand_int(get_fast_generator(), activation_layer_names.size());
+
+    // Name of new layer
+    std::string new_layer_name = "new_" + m_activation_types[type_index] 
+                                        + std::to_string(step) + std::to_string(name_index);
```

Comment:
If you are replacing the layer, it should be fine to reuse the old name. Right now we'll get layer names like `new_new_new_layer123`.

Reply:
Keeping layer name same.